graph/src/blockchain/mod.rs

```diff
@@ -264,6 +267,42 @@ pub trait TriggerData {
     fn error_context(&self) -> String;
 }
 
-pub trait MappingTrigger {
+pub trait MappingTrigger: Send + Sync {
     fn handler_name(&self) -> &str;
+
+    /// A flexible interface for writing a type to AS memory, any pointer can be returned.
+    /// Use `AscPtr::erased` to convert `AscPtr<T>` into `AscPtr<()>`.
+    fn to_asc<H: AscHeap>(self, heap: &mut H) -> Result<AscPtr<()>, DeterministicHostError>;
```

Comment:
What do you think about naming `to_asc_ptr` instead?

We have `to_asc_obj` and `to_asc_bytes`, by following the pattern I think this one should be specific as well.

Reply:
Renamed.